pages/apjktu/scheme2024/atp/atp-2.mdx

```diff
@@ -0,0 +1,117 @@
+# String operations using python
+
+## Aim
+
+Create, concatenate, and print a string and access a sub-string from a given string.
+
+## Algorithm
+
+1. Start
+2. Input a string from the user
+3. Take the choice input from the user
+    * Validate the choice input
+    * If the choice is 'e', exit the program
+    * If the choice is not one of the 3 choices, print "Invalid choice" and continue
+    * else, go to step 4
+4. Perform the operation
+    * If the choice is 1, concatenate the string
+    * If the choice is 2, access a substring
+    * If the choice is 3, print the string
+5. End
+
+## Program
+
+```python filename="string_operations.py" copy showLineNumbers
+# Function to concatenate two strings
+def concatenate_strings(string1, string2):
+    return string1 + string2
+
+# Function to derive a substring from a string
+def access_substring(string, start_index, end_index):
+    return string[start_index:end_index]
+
+# Input string
+string_1 = input("Enter a string: \n")
+
+while(True):
+    # Input the choice from the user
+    choice = input("Enter 1 to concatenate the string, 2 to access a substring, 3 to print the string \nor press 'e' to exit the program: \n")
+
+    # Validate the choice input
+    if choice.lower() == 'e':
+        break
+    elif choice not in ['1', '2', '3']:
+
+        print("Invalid choice")
+        continue
+    else:
+
+        # If the choice is 1, concatenate the string
+        if choice == '1':
+            string_2 = input("Enter another string: \n")
+            concatenated_string = concatenate_strings(string_1, string_2)
+            print(concatenated_string)
+
+        # If the choice is 2, access a substring
+        elif choice == '2':
+            start_index = int(input("Enter the start index: \n"))
+            end_index = int(input("Enter the end index: \n"))
+            sub_string = access_character(string_1, start_index, end_index)
```

Comment:
fix: change called function name from 'access_character' to 'access_substring'
reason: function call updation to reflect function name change

Reply:
Addressed